sync2/client.go

```diff
@@ -39,16 +41,30 @@ type HTTPClient struct {
 }
 
 func NewHTTPClient(shortTimeout, longTimeout time.Duration, destHomeServer string) *HTTPClient {
+	baseUrl := destHomeServer
+	if strings.HasPrefix(destHomeServer, "/") {
+	     baseUrl = "http://unix"
+	}
+
 	return &HTTPClient{
-		LongTimeoutClient: &http.Client{
-			Timeout:   longTimeout,
-			Transport: otelhttp.NewTransport(http.DefaultTransport),
-		},
-		Client: &http.Client{
-			Timeout:   shortTimeout,
-			Transport: otelhttp.NewTransport(http.DefaultTransport),
-		},
-		DestinationServer: destHomeServer,
+		LongTimeoutClient: newClient(longTimeout, destHomeServer),
+		Client:            newClient(shortTimeout, destHomeServer),
+		DestinationServer: baseUrl,
+	}
+}
+
+func newClient(timeout time.Duration, destHomeServer string) *http.Client {
+	transport := http.DefaultTransport
+	if strings.HasPrefix(destHomeServer, "/") {
+		transport = &http.Transport{
+			DialContext: func(_ context.Context, _, _ string) (net.Conn, error) {
+				return net.Dial("unix", destHomeServer)
```

Comment:
I dislike how unintuitive this is. Initially, we say "if it's unix, set the HS URL to `http://unix`" but the real magic comes from clobbering the transport function to dial the real file. Any poor developer reading this could easily think `destHomeServer == "http://unix"` and be very confused at this line.